tests/test_bokky_poo_bahs_date-time-library.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from brownie import chain, interface
+
+contract_addresses = {
+    1: "0x23d23d8f243e57d0b924bff3a3191078af325101",
+    3: "0x947cc35992e6723de50bf704828a01fd2d5d6641",
+}
+
+
+@pytest.mark.parametrize(
+    "timestamp,expected_date",
+    [
+        (0, (1970, 1, 1)),
+        (1709164800, (2024, 2, 29)),  # leap year
+        (1672531200, (2023, 1, 1)),  # 1 Jan start of the day
+        (1672617570, (2023, 1, 1)),  # 1 Jan end of the day
+        (1672531170, (2022, 12, 31)),  # 31 Dec end of the day
+    ],
+)
+def test_timestamp_to_date(date_time_library, timestamp, expected_date):
+    assert date_time_library.timestampToDate(timestamp) == expected_date
+
+
+@pytest.mark.parametrize(
+    "date, expected_timestamp",
+    [
+        ((1970, 1, 1), 0),
+        ((2024, 2, 29), 1709164800),  # 29 Feb start of the day of leap year
+        ((2023, 1, 1), 1672531200),  # 1 Jan start of the day
+        ((2022, 12, 31), 1672444800),  # 31 Dec start of the day
+    ],
+)
+# This method always returns timestamp of start of the day
+def test_timestamp_from_date(date_time_library, date, expected_timestamp):
+    assert date_time_library.timestampFromDate(*date) == expected_timestamp
+
+
+@pytest.mark.parametrize(
+    "start_timestamp, months_in_a_row, months_to_add",
+    [
+        [1577836800, 120, [1, 2, 3, 6, 8, 12, 24]],  # First day of month
```

Comment:
Consider refactoring to put for data in test parameters and leave no loops through a parameter
Also no need to  have `months_in_a_row` as a parameter, if it doesn't change